AutomataCLI/AutomataCLI/AutomataConverter.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using AutomataCLI;
+using AutomataCLI.Struct;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AutomataCLI {
+    public class AutomataConverter{
+        public class AutomataConverterException : Exception {
+            const String MESSAGE_BASE = "Invalid conversion. ";
+            public AutomataConverterException(String supplement) : base($"{MESSAGE_BASE}{supplement}") { }
+        }
+        public static Automata ToNFA(Automata convertedAutomata) {
+            
+            if(convertedAutomata.GetAutomataType() == AutomataType.AFN){
+                return convertedAutomata;
+            }
+
+            var newAutomata = new Automata();
+            newAutomata.AddSymbols(convertedAutomata.GetSymbols());
+            newAutomata.AddStates(convertedAutomata.GetStates());
+            newAutomata.AddTransitions(convertedAutomata.GetTransitions());
+
+            ConvertStatesToDFA(newAutomata, newAutomata.GetStates().ToList());
+            
+            return newAutomata;
+        }
+
+        public static void ConvertStatesToDFA(Automata automata, List<State> states) {
+
+            List<State> newStates = new List<State>();
+            foreach (State state in states) {
+                foreach (String symbol in automata.GetSymbols()) {
+                    var transitions = automata.GetTransitionsLike(state, symbol);
+
+                    if (transitions.Count == 0) continue;
+
+                    State newState;
+                    if (transitions.Count > 1) {
+                        newState = new GroupedState(transitions.Select(t => t.To).ToList());
+                        if(!automata.ContainsState(newState)){
+                            newStates.Add(newState);
+                            automata.AddState(newState);
+                        }
+                    } else {
+                        newState = transitions.First().To;
+                    }
+                    Transition newTransition = new Transition(state, symbol, newState);
+                    if(!automata.ContainsTransition(newTransition.From.Name, symbol, newTransition.To.Name)){
```

Comment:
You can pass the ´ContainsTransition´ method an instance (it will check if a Transition instance with the same stateFrom.Name, symbol and stateTo.Name already exists in the automata). 

```suggestion
                    if(!automata.ContainsTransition(newTransition)){
```